ls1intum/Artemis
# `Programming Exercises`: Truncate test name instead of silent fail

### Checklist
#### General


#### Server

### Motivation and Context


During the creation of an exercise for PGdP we noticed the absence of the results for the solution participation. After a bit of debugging, we noticed that Artemis silently fails without any error message, because the automatically generated test names for the parameterized tests were too long.

### Description

Artemis now truncates the test name to the maximum allowed length, if they are too long.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Log in to Artemis
2. Create a test with a long name (at least 256 characters, @DisplayName from JUnit might be easier than such a long method name). 
3. Check that the solution still gets a result in this exercise

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Constants.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5846/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.config/Constants.html) | 100% | ✅ |
| [ProgrammingExerciseTestCase.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5846/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/ProgrammingExerciseTestCase.html) | 98% | ✅ |
